docs/events/projections/multi-stream-projections.md

```diff
@@ -453,3 +453,94 @@ public class DayProjection: MultiStreamProjection<Day, int>
 ```
 <sup><a href='https://github.com/JasperFx/marten/blob/master/src/Marten.AsyncDaemon.Testing/ViewProjectionTests.cs#L124-L168' title='Snippet source file'>snippet source</a> | <a href='#snippet-sample_showing_fanout_rules' title='Start of snippet'>anchor</a></sup>
 <!-- endSnippet -->
+
+## Using Custom Grouper with Fan Out Feature for Event Projections
+
+In Marten, the `MultiStreamProjection` feature allows for complex transformations and aggregations of events. However, there might be scenarios where a single event in your domain carries information that is more suitable to be distributed across multiple instances of your projected read model. This is where the combination of a Custom Grouper and the Fan Out feature comes into play.
+
+### The Scenario
+
+Imagine you have a system where `EmployeeAllocated` events contain a list of allocations for specific days. The goal is to project this information into a monthly summary. 
+
+### Custom Projection with Custom Grouper
+
+The `MonthlyAllocationProjection` class uses a custom grouper for this transformation. Here, `TransformsEvent<EmployeeAllocated>()` indicates that events of type `EmployeeAllocated` will be used even if there are no direct handlers for this event type in the projection.
+
+<!-- snippet: snippet-sample_showing_registering_custom_grouper_with_fan_out -->
+<a id='snippet-sample_showing_registering_custom_grouper_with_fan_out'></a>
```

Comment:
**FIX:** Marten gets snippets automatically thanks to [MarkdownSnippets](https://github.com/SimonCropp/MarkdownSnippets). To have such, you need to find the source code (usually located in tests) and wrap it with the exact region name and use it in docs with the `snippet-` prefix. The code sample is located here: https://github.com/JasperFx/marten/blob/4dafe48f4209a90b5c66ee8a9a2a60e15d7ca991/src/EventSourcingTests/Projections/MultiStreamProjections/CustomGroupers/custom_grouper_with_events_transformation.cs#L41.

Reply:
I'm not 100% sure if I did it correct now

Reply:
Should i add regions for the small blocks as code in the source as well, instead of copying them to the snippets in the docs and let tooling do the copying for me?
Like I tried in commit: https://github.com/JasperFx/marten/pull/2695/commits/062620a4b58c2f403d90bf0b333b286f235f862c